### PR TITLE
Decouple authentication

### PR DESCRIPTION
**Pull request by MaSer713**
Changing the authentication and registration to work with microservices. There are 3 public services right now: registration, authentication and token. 
The first one is consumed by the client to create a new account.
The second one is consumed by the client to exchange credentials for an authentication token.
The third one is consumed by the server to verify the authentication token received from the server (this should probably be replaced with local on-server verification later on).

Old registration/authentication handling on the server are gone.

### CODE REVIEW

File: Server/src/server/dataCenter/DataCenter.java [hunk 1]
```diff
@@ -98,65 +98,71 @@ public String getClientName(String username) {
...
+            accounts.replace(account, client);
+            clients.replace(client, account);
+            GameServer.addToSendingMessages(Message.makeAccountCopyMessage(client, account));
+            GameServer.serverPrint(client + " Is Logged In");
+        }
+    }
+
+    private void register(String username, String client) throws LogicException {
+        if (username == null || username.length() < 2
+                || getAccount(username) != null) {
```
Comment (fluffy-hamster) [inline Server/src/server/dataCenter/DataCenter.java]: When people register, is there are database somewhere? 

If so, do we not also need to check in name has not been taken?

File: Shared/src/shared/models/services/WebApiRequest.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package shared.models.services;
+
+public interface WebApiRequest {
```
Comment (fluffy-hamster) [inline Shared/src/shared/models/services/WebApiRequest.java]: Is there much point in adding empty interfaces?   It just makes the code messy until someone gets round to doing something with them.

File: Client/src/main/java/org/projectcardboard/client/controller/Client.java [hunk 3]
```diff
@@ -33,6 +34,7 @@
     private Show currentShow;
     private final Gson gson = new Gson();
     private static final String serverName = Config.getInstance().getProperty("SERVER_NAME");
+    private static final String GENERIC_ERROR = "Unknown error";
```
[the remark this answers is not in the document]
Comment (MaSer713) [reply Client/src/main/java/org/projectcardboard/client/controller/Client.java]: It is used whenever the error message is null, therefore - unknown error.

File: Shared/src/shared/models/services/WebApiRequest.java (hunk 2 above)
Comment (MaSer713) [reply Shared/src/shared/models/services/WebApiRequest.java]: It is empty on purpose and will probably stay this way indefinitely. This is what ApiClientService accepts to send over.

File: Server/src/server/dataCenter/DataCenter.java (hunk 1 above)
Comment (MaSer713) [reply Server/src/server/dataCenter/DataCenter.java]: Yes, and it is being checked both in this repo and the microservice doing the actual registration.

File: Client/src/main/java/org/projectcardboard/client/services/PasswordConverterService.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+public final class PasswordConverterService {
+
+    private MessageDigest digest;
+
+    public PasswordConverterService(MessageDigest digest) {
+        this.digest = digest;
+    }
+
+    public String convert(String login, String password) {
```
[the remark this answers is not in the document]
Comment (MaSer713) [reply Client/src/main/java/org/projectcardboard/client/services/PasswordConverterService.java]: Done

**Pull request merged**